File: gymcrm.web/src/modules/identity/types/insertAccount.ts
// Mirrors GymCRM.IdentityAPI.Models.DTOs.InsertAccount 1:1.
// Body for POST /Authentication/Register.
export interface InsertAccount {
    email: string;
    password: string;
    accountType?: number;
    gymSubscriptionType?: number;
    gender?: number;
}

File: GymCRM.IdentityAPI/Models/DTOs/InsertAccount.cs
using System.ComponentModel.DataAnnotations;

namespace GymCRM.IdentityAPI.Models.DTOs;

public class InsertAccount
{
    [Required]
    public string Email { get; set; }
    [Required]
    public string Password { get; set; }
    public int? AccountType { get; set; }
    public int? Gender { get; set; }
    public string? TimeZone { get; set; }
}